pvlib/iotools/pvgis.py

```diff
@@ -452,22 +457,34 @@ def get_pvgis_tmy(lat, lon, outputformat='json', usehorizon=True,
     data = None, None, None, None
     if outputformat == 'json':
         src = res.json()
-        return _parse_pvgis_tmy_json(src)
+        data, months_selected, inputs, meta = _parse_pvgis_tmy_json(src)
     elif outputformat == 'csv':
         with io.BytesIO(res.content) as src:
-            data = _parse_pvgis_tmy_csv(src)
+            data, months_selected, inputs, meta = _parse_pvgis_tmy_csv(src)
     elif outputformat == 'basic':
         with io.BytesIO(res.content) as src:
-            data = _parse_pvgis_tmy_basic(src)
+            data, months_selected, inputs, meta = _parse_pvgis_tmy_basic(src)
     elif outputformat == 'epw':
         with io.StringIO(res.content.decode('utf-8')) as src:
             data, meta = parse_epw(src)
-            data = (data, None, None, meta)
+            months_selected, inputs = None, None
     else:
         # this line is never reached because if outputformat is not valid then
         # the response is HTTP/1.1 400 BAD REQUEST which is handled earlier
         pass
-    return data
+
+    if map_variables is None:
+        warnings.warn(
```

Comment:
We need to test that this warning is emitted and mark that test with the [`fail_on_pvlib_version`](https://github.com/pvlib/pvlib-python/blob/0fd5c43acc8a90972bf5018d3c07b6ad8b93a5fd/pvlib/tests/conftest.py#L21) decorator. This ensures that we don't forget to remove deprecated functionality.

Reply:
@who Very smart, thanks for showing me this!

I believe all of your comments have been addressed.